middleware.ts

```diff
@@ -31,14 +31,16 @@ export async function middleware(request: NextRequest) {
     pathname.startsWith("/static") ||
     pathname === "/favicon.ico" ||
     pathname.startsWith("/auth/callback") ||
-    pathname.startsWith("/onboarding") ||
     pathname === "/"
   ) {
     return NextResponse.next();
   }
 
-  // Guard restricted routes under /app
-  if (pathname.startsWith("/app")) {
+  // General guard for /app and /onboarding
+  const isOnboarding = pathname.startsWith("/onboarding");
+  const isApp = pathname.startsWith("/app");
+
+  if (isOnboarding || isApp) {
```

Comment:
```suggestion
  const isApp = pathname.startsWith("/app");
  const isDashboard = pathname.startsWith("/dashboard");

  if (isOnboarding || isApp || isDashboard) {
```

Would this also solve the need to protect the Dashboard page route?

Reply:
yes thank you!